### ISSUE TITLE
support for Plot 0.6

### ISSUE THREAD

**Issue by Fil** [umwelt-data/olli#33]
Hello, thank you for supporting Observable Plot! I've published an [Observable notebook](https://observablehq.com/@fil/hello-olli) to demonstrate using olli with Plot.

Note that we've changed the internal representation of channels between Plot 0.5 and Plot 0.6 — the following addition to packages/adapters/src/ObservablePlotAdapter.ts should help to support both versions:

```diff
diff --git a/packages/adapters/src/ObservablePlotAdapter.ts b/packages/adapters/src/ObservablePlotAdapter.ts
index 0c003bd..d3450da 100644
--- a/packages/adapters/src/ObservablePlotAdapter.ts
+++ b/packages/adapters/src/ObservablePlotAdapter.ts
@@ -45,7 +45,7 @@ function plotToFacetedChart(plot: any, svg: Element): FacetedChart {
         plot.facet.y ?
             plot.facet.y :
             plot.facet.x :
-        plot.marks.find((mark: any) => mark.ariaLabel === 'line').channels.find((c: any) => c.name === "stroke").value;
+        flatChannels(plot.marks.find((mark: any) => mark.ariaLabel === 'line').channels).find((c: any) => c.name === "stroke").value;
     if (hasFacets(plot)) {
         charts = new Map(Object.values(chartSVG.children)
             .filter((n) => n.getAttribute('aria-label') === 'facet')
@@ -112,7 +112,7 @@ function plotToChart(plot: any, svg: Element): Chart {
 function parseAxis(plot: any, svg: Element): Axis {
     const axisType = svg?.getAttribute('aria-label') === 'y-axis' ? 'y' : 'x'
     const plotMark = plot.marks.filter((mark: any) => mark.ariaLabel !== 'rule')[0]
-    const channel = plotMark.channels.find((c: any) => c.scale === axisType)
+    const channel = flatChannels(plotMark.channels).find((c: any) => c.scale === axisType)
     const field: string = typeof channel.value === 'object' ? channel.value.label : channel.value
     const ticks: string[] = Object.keys(svg.children).map((k: string) => {
         const cObj: Element = svg.children[parseInt(k)]
@@ -152,7 +152,7 @@ function parseAxis(plot: any, svg: Element): Axis {
  */
 function parseLegend(plot: any, svg: Element): Legend { //TODO: Does not support 'ramp' legend types when the legend is rendered as an SVG
     const plotMark = plot.marks.filter((mark: any) => mark.ariaLabel !== 'rule')[0];
-    const channel = plotMark.channels.find((c: any) => c.scale === 'color'); // TODO channel hardcoded to color
+    const channel = flatChannels(plotMark.channels).find((c: any) => c.scale === 'color'); // TODO channel hardcoded to color
     const values: string[] = Object.keys(svg.children).map((k: string) => {
         let c = svg.children[parseInt(k)];
         if (c.nodeName !== 'STYLE') {
@@ -226,3 +226,7 @@ function plotMarkToOlliMark(m: string): OlliMark | undefined {
             return undefined
     }
 }
+
+function flatChannels(channels: any) {
+    return channels.find ? channels : Object.entries(channels).map(([name, chan]) => ({...chan as {}, name}))
+}

```

Hope this helps. As you can see I still have to find my way in this project — I'll try and contribute in a more efficient manner next time. I'm also happy to answer any question re: Plot, and open to suggestions that could help bridge the gap.

**Comment by jonathanzong** [comment]
Hi Fil, thanks so much for taking the time to do all of this! I've added a link to your example notebook to the quickstart docs on the website, and made your suggested changes to the adapter. I'll plan to publish the patch within the next day or so, pending a few other fixes I want to make too.

Really appreciate your interest, and I hope we can chat a bit more about the kinds of information Olli needs for accessibility, and potentially making it easier to surface these things in the Plot API. I'm hoping to add more documentation about adapters soon too, which hopefully will help structure that conversation.

**Comment by jonathanzong** [comment]
fixed in 3d1b4e0f4131da85a4dd3e74d3c7d00b9033d3bb – 7f4d8409dd9dd1026a34ce12cf7bf2ada1903a4a
